fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java

```diff
@@ -5000,7 +5001,10 @@ public boolean unprotectDropTable(Database db, long tableId, boolean isForceDrop
 
         db.dropTable(table.getName());
         if (!isForceDrop) {
-            Catalog.getCurrentRecycleBin().recycleTable(db.getId(), table);
+            Table oldTable = Catalog.getCurrentRecycleBin().recycleTable(db.getId(), table);
+            if (oldTable != null && oldTable.getType() == TableType.OLAP) {
+                Catalog.getCurrentCatalog().onEraseOlapTable((OlapTable) oldTable, false);
```

Comment:
`unprotectDropTable` is called by `dropTable` with the database write lock.
This will block almost all operations in the database.